contracts/strategies/StrategyBase.sol

```diff
@@ -0,0 +1,298 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+pragma experimental ABIEncoderV2;
+
+// Hardhat
+import "hardhat/console.sol";
+
+// Lyra
+import {VaultAdapter} from "@lyrafinance/core/contracts/periphery/VaultAdapter.sol";
+import {GWAVOracle} from "@lyrafinance/core/contracts/periphery/GWAVOracle.sol";
+
+// Libraries
+import {Vault} from "../libraries/Vault.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {LyraVault} from "../core/LyraVault.sol";
+import {DecimalMath} from "@lyrafinance/core/contracts/synthetix/DecimalMath.sol";
+import {SignedDecimalMath} from "@lyrafinance/core/contracts/synthetix/SignedDecimalMath.sol";
+
+contract StrategyBase is VaultAdapter {
+  using DecimalMath for uint;
+  using SignedDecimalMath for int;
+
+  LyraVault public immutable vault;
+  OptionType public immutable optionType;
+  GWAVOracle public immutable gwavOracle;
+
+  /// @dev asset used as collateral in AMM to sell. Should be the same as vault asset
+  IERC20 public collateralAsset;
+
+  mapping(uint => uint) public lastTradeTimestamp;
+
+  uint public activeExpiry;
+  uint[] public activeStrikeIds;
+  mapping(uint => uint) public strikeToPositionId;
+
+  // basic strategy parameters
+  struct BaseStrategy {
```

Comment:
No strong opinion on this, but I would try and make the StrategyBase.sol as generic as possible. Maybe we don't have a base strategy struct and let the integrator decide what bounds to use.

Reply:
So the only framework we give integrators is: 
(1) you can open multiple strikes/positions 
(2) you can open/close positions
(3) you can track capital used/received
(4) anyone can ping vault to open/close but you can create whatever logic you want to determine whether to allow that action.